api/src/main/java/org/openmrs/module/locationbasedaccess/PatientControlImpl.java

```diff
@@ -0,0 +1,167 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+
+package org.openmrs.module.locationbasedaccess;
+
+
+import org.aopalliance.aop.Advice;
+import org.aopalliance.intercept.MethodInvocation;
+import org.aopalliance.intercept.MethodInterceptor;
+import org.openmrs.Patient;
+import org.openmrs.Person;
+import org.openmrs.PersonAttribute;
+import org.openmrs.User;
+import org.openmrs.Location;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.locationbasedaccess.common.Constants;
+import org.springframework.aop.Advisor;
+import org.springframework.aop.framework.ReflectiveMethodInvocation;
+import org.springframework.aop.support.StaticMethodMatcherPointcutAdvisor;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Set;
+
+/**
+ * It will used to implement the AOP Methods for PatientServiceImpl class
+ */
+public class PatientControlImpl extends StaticMethodMatcherPointcutAdvisor implements Advisor {
```

Comment:
Shouldn't this be named PersonServiceAdvisor?

Reply:
I will change it to Persons and will update the PR content as I addressed above